core/core-impl/src/main/java/io/fabric8/launcher/core/impl/documentation/BoosterDocumentationStoreImpl.java

```diff
@@ -0,0 +1,127 @@
+        this.async = requireNonNull(async, "async must be specified");
+        this.downloader = requireNonNull(downloader, "downloader must be specified");
+    }
+
+    // Initialize on startup
+    public void init(@Observes @Initialized(ApplicationScoped.class) Object init) {
+        // Do nothing
+    }
+
+    @PostConstruct
+    public void initialize() {
+        getDocumentationPath();
+    }
+
+    @Override
+    public CompletableFuture<Path> reloadDocumentation() {
+        return getDocumentationPath(true);
+    }
+
+    @Override
+    public synchronized CompletableFuture<Path> getDocumentationPath() {
+        return getDocumentationPath(false);
+    }
+
+    @Override
+    public void waitForDocumentation() throws ExecutionException, InterruptedException {
+        getDocumentationPath().get();
+    }
+
+    private synchronized CompletableFuture<Path> getDocumentationPath(final boolean reload) {
+        if (reload || pathCompletableFuture == null) {
+            pathCompletableFuture = createDocumentationPathFuture();
+        }
+        return pathCompletableFuture;
+    }
+
+    private CompletableFuture<Path> createDocumentationPathFuture() {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                return downloader.download();
+            } catch (final IOException e) {
+                logger.log(Level.SEVERE, "Error while loading documentation", e);
+                throw new IllegalStateException("Error while loading documentation", e);
+            }
+        }, async);
+    }
+
+    private static Path cloneGitRepository(final String readmeRepositoryURI, final String branch) throws IOException {
+        logger.log(Level.INFO, "Indexing contents from {0} using {1} ref",
+                   new Object[]{readmeRepositoryURI, branch});
+        final Path catalogPath = Files.createTempDirectory("booster-readme");
+        logger.info("Created " + catalogPath);
+        ProcessBuilder builder = new ProcessBuilder()
+                .command("git", "clone", readmeRepositoryURI,
+                         "--branch", branch,
+                         "--recursive",
+                         "--depth=1",
+                         "--quiet",
+                         "-c", "advice.detachedHead=false",
+                         catalogPath.toString())
+                .inheritIO();
+        logger.info("Executing: " + builder.command().stream().collect(Collectors.joining(" ")));
+        try {
+            int exitCode = builder.start().waitFor();
+            assert exitCode == 0 : "Process returned exit code: " + exitCode;
+        } catch (InterruptedException e) {
+            logger.log(Level.WARNING, "Interrupted indexing process");
+            throw new IOException("Interrupted", e);
+        }
+        return catalogPath;
+    }
+
+    public interface DocumentationFilesDownloader {
```

Comment:
Do you think this interface is needed since it's only used in testing or can it be replaced by a `Supplier<Path>`?

Reply:
I was thinking of using this one in a later commit to use a local directory directly (if env is set), so I do prefer the second solution ;)